### ISSUE TITLE
Failed to launch on M2 2023 running Sonoma

### ISSUE THREAD

**Issue by jamieleepreece** [navtoj/NotchBar#12]
I've tried to launch from finder and terminal and keep getting the "The application 'NotchBar' can't be opened".

I've rebooted and tried to unarchive via the CLI and alternative archivers, but no avail. Here is the message from terminal
```
The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x6000021d43c0 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
```

**Comment by navtoj** [comment]
Have you tried the macOS built-in `Archive Utility` app?

https://support.apple.com/en-ca/guide/mac-help/mchlp2528/14.0/mac/14.0

![Screenshot 2024-10-11 at 12 17 17 PM](https://github.com/user-attachments/assets/639a41f7-29da-4341-97b0-cb657effb2c5)

**Comment by jamieleepreece** [comment]
Sorry, I can unzip just fine. The problem is launching as stated above e.g.
![Screenshot 2024-10-13 at 19 23 37](https://github.com/user-attachments/assets/93225375-1313-47eb-a016-e62960da5c08)

Also tried to execute from within the package contents, but no avail.

**Comment by jamieleepreece** [comment]
Changed permissions for NotchBar in `NotchBar.app/Contents/MacOS/NotchBar` and can now launch via terminal. I'm currently 'docked', so getting this warning "NotchBar only supports devices with a notch."

**Comment by navtoj** [comment]
> Changed permissions for NotchBar in `NotchBar.app/Contents/MacOS/NotchBar` and can now launch via terminal.

Looks like `.zip` is breaking file permissions for the executable within the `.app` bundle. I'll replace `.zip` with `.aar` ([Apple Archive Format](https://developer.apple.com/documentation/applearchive)) going forward.

> I'm currently 'docked', so getting this warning "NotchBar only supports devices with a notch."

Screenshot please. Also, I hadn't accounted for this use case but it's easy to enable.

When the MacBook screen is closed, where do you envision the NotchBar to show on your other screen?

**Comment by jamieleepreece** [comment]
1. Personally I would like to see the NotchBar app only on the MBP notched screen. However, depending on how useful the app is, it *could* be be used on non-notched screens as well. Perhaps by default it shows only on the notched screen, but with a menu item (via icon) to enable for all/primary screens. When connected to an external monitor for example, I would expect the MBP screen to only show the app, as the external screens do not have notches and are unaffected.
I always configure my OSX to be visually minimalist. For example, I hide the app bar and the menu bar, unless the mouse hovers near, or I open the menu via keyboard. This is the main issue for me with the new MBPs and the notch – I can't have full control of the screen and have a white/grey bar at the top distracting me, unless I use a black desktop background (anti consumer). I use the MBP freely, as well as docked throughout the day, which is why I think having support for external screens may be useful. However, I think having it only display on notched screens is fine for now. 
2. Whenever possible I do not use any apps that persist on the screen as a distraction. This app is a compromise for me, as Apple have made it very difficult to work with the entire screen in this case. In this case I'd rather have that space utilised than just appear as a constant black background.

**Comment by navtoj** [comment]
Thanks so much for the detailed response.

I completely share your thoughts on this. I had the same exact issue with the empty space around the notch so ended up starting this project.

Regarding support for external monitors, it will likely come after the baseline features for notched screens have been added.

> I'm currently 'docked', so getting this warning "NotchBar only supports devices with a notch."

Also, if possible, could I please get a screenshot of the warning above because I can't find this anywhere in my code so really confused where it's coming from. Thanks.

**Comment by navtoj** [comment]
Patched: 77f05cf56351b3d6b27b07ea50d33e311292ba36

**Comment by jamieleepreece** [comment]
The terminal displays the message. However, launching the app via GUI doesn't give any feedback
![Screenshot 2024-10-15 at 10 22 53](https://github.com/user-attachments/assets/eafcac32-3802-430e-9e97-68e4a26c2a93)
